backend/analysis-engine/src/main/java/org/sonarsource/sonarlint/core/analysis/command/Command.java

```diff
@@ -39,4 +40,14 @@ public boolean isReady() {
   public void cancel() {
     // most commands are not cancelable
   }
+
+  public boolean shouldCancel(Command executingCommand) {
+    if (!(this instanceof AnalyzeCommand analyzeCommandA) || !(executingCommand instanceof AnalyzeCommand analyzeCommandB)) {
+      return false;
+    }
+    var triggerTypesMatch = analyzeCommandA.getTriggerType() == analyzeCommandB.getTriggerType();
+    var filesMatch = Objects.equals(analyzeCommandA.getFiles(), analyzeCommandB.getFiles());
+    var extraPropertiesMatch = Objects.equals(analyzeCommandA.getExtraProperties(), analyzeCommandB.getExtraProperties());
+    return triggerTypesMatch && filesMatch && extraPropertiesMatch;
+  }
```

Comment:
Nitpick: You could have a default implementation in `Command` that always return `false` and have an overridden implementation in `AnalyzeCommand`. This avoids 1 small check, so not a big deal